packages/next-swc/crates/next-core/src/app_segment_config.rs

```diff
@@ -235,11 +245,26 @@ fn parse_config_value(
         }
         "revalidate" => {
             let value = eval_context.eval(init);
-            let Some(val) = value.as_bool() else {
-                return invalid_config("`revalidate` needs to be a static boolean", &value);
-            };
-
-            config.revalidate = val;
+            match value {
+                JsValue::Constant(ConstantValue::Num(ConstantNumber(val))) if val > 0.0 => {
```

Comment:
Is this handling numeric separators? x-ref: https://github.com/vercel/next.js/pull/51438

Reply:
I think so. SWC should handle that way earlier.